src/util/config.rs

```diff
@@ -167,6 +273,182 @@ pub fn parse_store_labels(labels: &str) -> Result<HashMap<String, String>, Confi
     Ok(map)
 }
 
+pub struct ReadableSize(pub u64);
+
+impl Serialize for ReadableSize {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+        where S: Serializer
+    {
+        let size = self.0;
+        let mut buffer = String::new();
+        if size % PB == 0 {
+            write!(buffer, "{}PB", size / PB).unwrap();
+        } else if size % TB == 0 {
+            write!(buffer, "{}TB", size / TB).unwrap();
+        } else if size % GB as u64 == 0 {
+            write!(buffer, "{}GB", size / GB).unwrap();
+        } else if size % MB as u64 == 0 {
+            write!(buffer, "{}MB", size / MB).unwrap();
+        } else if size % KB as u64 == 0 {
+            write!(buffer, "{}KB", size / KB).unwrap();
+        } else {
+            return serializer.serialize_u64(size);
+        }
+        serializer.serialize_str(&buffer)
+    }
+}
+
+impl<'de> Deserialize<'de> for ReadableSize {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+        where D: Deserializer<'de>
+    {
+        struct SizeVisitor;
+
+        impl<'de> Visitor<'de> for SizeVisitor {
+            type Value = ReadableSize;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str("valid size")
+            }
+
+            fn visit_i64<E>(self, size: i64) -> Result<ReadableSize, E>
+                where E: de::Error
+            {
+                if size >= 0 {
+                    self.visit_u64(size as u64)
+                } else {
+                    Err(E::invalid_value(Unexpected::Signed(size), &self))
+                }
+            }
+
+            fn visit_u64<E>(self, size: u64) -> Result<ReadableSize, E>
+                where E: de::Error
+            {
+                Ok(ReadableSize(size))
+            }
+
+            fn visit_str<E>(self, size_str: &str) -> Result<ReadableSize, E>
+                where E: de::Error
+            {
+                let err_msg = "valid size, only KB, MB, GB, TB, PB are supported.";
+                let size_str = size_str.trim();
+                if size_str.len() < 2 {
+                    return Err(E::invalid_value(Unexpected::Str(size_str), &err_msg));
+                }
+                
+                let (number_str, unit_str) = size_str.split_at(size_str.len() - 2);
+                let unit = match &*unit_str.to_lowercase() {
```

Comment:
In most cases, we will treat kb and KB as the same unit. I think we don't need to distinguish them.

Btw, I still prefer KB instead of K, MB instead of M.